lib/site_prism/dsl.rb

```diff
@@ -36,6 +36,14 @@ def raise_if_block(obj, name, has_block, type)
       raise SitePrism::UnsupportedBlockError
     end
 
+    # Prevent users from naming the elements starting with no_
+    def raise_if_name_unsupported(obj, name)
+      if name.to_s.start_with?("no_")
+        SitePrism.logger.error("#{obj.class}#{name} should not start with no_")
+        raise SitePrism::UnsupportedElementName
```

Comment:
can you look inside the SitePrism::Deprecator class. In there I would like this to be raised as a hard deprecation.

https://github.com/site-prism/site_prism/blob/master/lib/site_prism/deprecator.rb#L6

It takes 2 arguments, but the 2nd one is optional. Up to you how you word it. But we cannot add this behaviour before v4.